File: js/src/maplestory/api/dto/character/characterAndroidEquipmentSkinDto.ts
import {
  CharacterAndroidEquipmentSkinBody,
} from '../../response/character/characterAndroidEquipmentBody';

/**
 * 캐릭터 안드로이드 피부 정보
 */
class CharacterAndroidEquipmentSkinDto {
  /**
   * 피부 명
   */
  skinName: string;

  /**
   * 색상 계열
   */
  colorStyle: string | null;

  /**
   * 피부 색조
   */
  hue: number | null;

  /**
   * 피부 채도
   */
  saturation: number | null;

  /**
   * 피부 명도
   */
  brightness: number | null;

  constructor(obj: CharacterAndroidEquipmentSkinBody) {
    const { skin_name, color_style, hue, saturation, brightness } = obj;

    this.skinName = skin_name;
    this.colorStyle = color_style;
    this.hue = hue;
    this.saturation = saturation;
    this.brightness = brightness;
  }
}

export { CharacterAndroidEquipmentSkinDto };

File: java/src/main/java/dev/spiralmoon/maplestory/api/tms/dto/character/CharacterAndroidEquipmentSkinDTO.java
package dev.spiralmoon.maplestory.api.tms.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

/**
 * 機器人膚色資訊
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterAndroidEquipmentSkinDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterAndroidEquipmentSkinDTO {

    /**
     * 膚色名稱
     */
    @JsonProperty("skin_name")
    private String skinName;

    /**
     * 色系
     */
    @JsonProperty("color_style")
    private String colorStyle;

    /**
     * 膚色色調
     */
    @JsonProperty("hue")
    private Integer hue;

    /**
     * 膚色彩度
     */
    @JsonProperty("saturation")
    private Integer saturation;

    /**
     * 膚色明度
     */
    @JsonProperty("brightness")
    private Integer brightness;
}
